File: leet-code/455. Assign Cookies/AssignCookies455_3.java
package com.example;

import java.util.Arrays;

public class AssignCookies455_3 {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int count = 0;
        int gIdx = 0;
        for (int size : s) {
            if (gIdx == g.length) break;
            if (size < g[gIdx]) continue;
            gIdx += 1;
            count += 1;
        }
        return count;
    }
}

File: leet-code/455. Assign Cookies/AssignCookies455_3.kt
package com.example

class AssignCookies455_3 {
    fun findContentChildren(g: IntArray, s: IntArray): Int {
        g.sort()
        s.sort()
        var i = 0
        var j = 0
        var count = 0
        while (i <= g.lastIndex && j <= s.lastIndex) {
            if (g[i] <= s[j]) {
                count += 1
                i += 1
            }
            j += 1
        }
        return count
    }
}
